client/transaction-pool/src/lib.rs

```diff
@@ -130,3 +175,158 @@ impl<PoolApi, Block> TransactionPool for BasicPool<PoolApi, Block>
 		self.pool.on_broadcasted(propagations)
 	}
 }
+
+#[cfg_attr(test, derive(Debug))]
+enum RevalidationStatus<N> {
+	/// The revalidation has never been completed.
+	NotScheduled,
+	/// The revalidation is scheduled.
+	Scheduled(Option<std::time::Instant>, Option<N>),
+	/// The revalidation is in progress.
+	InProgress,
+}
+
+enum RevalidationStrategy<N> {
+	Always,
+	Light(RevalidationStatus<N>)
+}
+
+impl<N: Clone + Copy + SimpleArithmetic> RevalidationStrategy<N> {
+	pub fn clear(&mut self) {
+		if let Self::Light(status) = self { status.clear() }
```

Comment:
```suggestion
		if let Self::Light(status) = self {
			status.clear()
		}
```

Reply:
Don't think it's addressed. I really find it ugly to keep it in one line.

Reply:
ops sorry!